### PR TITLE
Register style handle on non-existing block stylesheets

### PR DESCRIPTION
**Pull request by walbo**
Trac ticket: https://core.trac.wordpress.org/ticket/53375

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/blocks.php [hunk 1]
```diff
@@ -151,23 +151,27 @@ function register_block_style_handle( $metadata, $field_name ) {
...
-	if ( file_exists( str_replace( '.css', '-rtl.css', $style_file ) ) ) {
-		wp_style_add_data( $style_handle, 'rtl', 'replace' );
-	}
+
 	if ( file_exists( $style_file ) ) {
+		$result = wp_register_style(
+			$style_handle,
+			$style_uri,
+			array(),
+			filemtime( $style_file )
```
Comment (desrosj) [inline src/wp-includes/blocks.php:160]: Ideally, we should not use `filetime()` for production sites. Is there a reason why we can't use the WordPress version here? When `version` is omitted here, the default should be the current WordPress Core version.

File: src/wp-includes/blocks.php [hunk 2]
```diff
@@ -151,23 +151,27 @@ function register_block_style_handle( $metadata, $field_name ) {
...
-		$style_handle,
-		$style_uri,
-		array(),
-		$version
-	);
-	if ( file_exists( str_replace( '.css', '-rtl.css', $style_file ) ) ) {
-		wp_style_add_data( $style_handle, 'rtl', 'replace' );
-	}
+
 	if ( file_exists( $style_file ) ) {
```
Comment (desrosj) [inline src/wp-includes/blocks.php:155]: I don't love these `file_exists()` checks. Is there any way to determine whether a file is present based on the block's metadata?

File: src/wp-includes/blocks.php (hunk 1 above)
Comment (desrosj) [reply src/wp-includes/blocks.php:160]: Can block metadata pass a `version`? I can't find the metadata documented anywhere within the inline documentation, so it's hard to know for sure what will be included.

As a spin-off, I'd love to get at least the required fields, and maybe the officially recognized optional ones documented.

Comment (gziolo) [reply src/wp-includes/blocks.php:160]: It was the case before, so we should keep the existing handling.

Comment (gziolo) [reply src/wp-includes/blocks.php:160]: Block metadata is documented at https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#block-api

Should we link it in a particular place in the code?

There is no `version` in the definition. For JavaScript files, we read additional data like the `version` (unique hash based on the file content) from the `.asset.php` file generated by webpack. We don't have anything like that for CSS files, but we might consider extending the build functionality to align with JS.

**Comment by walbo** [comment]
Closing in favor of #1370

**Pull request closed without merging**